api/SubmitJob/cloudformation.yml

```diff
@@ -0,0 +1,63 @@
+AWSTemplateFormatVersion: 2010-09-09
+
+Outputs:
+
+  LambdaArn:
+    Value: !GetAtt Lambda.Arn
+
+Parameters:
+  JobQueueArn:
+    Type: String
+
+  JobDefinition:
+    Type: String
+
+Resources:
+
+  LogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub "/aws/lambda/${AWS::StackName}"
+      RetentionInDays: 90
+
+  Role:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Ref AWS::StackName
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          Action: sts:AssumeRole
+          Principal:
+            Service: lambda.amazonaws.com
+          Effect: Allow
+      Policies:
+      - PolicyName: policy
+        PolicyDocument:
+          Version: 2012-10-17
+          Statement:
+          - Effect: Allow
+            Action:
+            - logs:CreateLogStream
+            - logs:PutLogEvents
+            Resource: !GetAtt LogGroup.Arn
+          - Effect: Allow
+            Action: batch:SubmitJob
+            Resource:
+              - !Ref JobQueueArn
+              - !Ref JobDefinition
+
+  Lambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      FunctionName: !Ref AWS::StackName
+      Environment:
+        Variables:
+          JobQueue: !Ref JobQueueArn
+          JobDefinition: !Ref JobDefinition
```

Comment:
UPPER_CASE for environment variable names?  don't feel too strongly about it

I also prefer to be specific about whether values are Names or ARNs, `JOB_QUEUE_ARN` and `JOB_DEFINITION_ARN` would be preferable.

I don't think we need it here, but for GRFN I had times where I wanted structured config (which you can't get with just environment variables), so I had a single environment variable with a big json string:

```
Environment:
  Variables:
    Config: !Sub |-
      {
        "job_queue": ${JobQueueArn},
        "job_definition": ${JobDefinition}
      }
```
then `config = json.load(environ['CONFIG'])` or something in the lambda

Reply:
I think that config trick could be more usefull if we need to inject more variables, But for how it is now I think this is cleaner.